tests/testsuite/cargo_add_with_vendored_pkgs.rs

```diff
@@ -0,0 +1,52 @@
+use cargo_test_support::{project, registry::Package};
+
+#[cargo_test]
+fn carg_add_with_vendored_packages() {
```

Comment:
Please put this in with the other cargo add tests (`tests/testsuite/cargo_add`) and look to the tests in there for how they should be written (its using the document UI test pattern)